### ISSUE TITLE
Editing a question takes it out of the collection

### ISSUE THREAD

**Issue by ricardojmendez** [metabase/metabase#5453]
To reproduce:

- Edit a question that's in a collection
- Change the database (for instance, if you're switching from a live to a reporting database)
- Save while Replacing the original question

The new question will not be on the same collection as the original one, will not have one at all.

**Comment by camsaul** [comment]
Yikes

@metabase/core-developers @attekei 
The backend assumes if you pass in `:collection_id nil` to the edit Question endpoint that you want to remove it from its collection. I'm guessing that's what's happening. We need to either:

1.  Make sure we don't pass a `:collection_id` key when we don't want to modify it (the backend will only remove a Question from a collection if `:collection_id` is passed as `:null`), or
2.  Make sure when we call the endpoint we're also passing the original `:collection_id` as well.

Basically the idea is either pass the existing value back or don't pass it at all, but whatever you do *don't* pass a different value back, which is what it sounds like is happening. This must be some weird edge case that's happening when the database is switched (perhaps some state containing `:collection_id` is getting cleared out?)
